### PR TITLE
Add eslint config for vue and es6 imports

### PR DESCRIPTION
**Pull request by jonboiser**
This adds an ESlint Config with the basic checks

* eslint:Recommended - for potential bugs and other things
* vue - potential bugs, plus conventions on directives, and order of props
* import - potential bugs in `import` statements. had to configure it to use node module resolution algorithm and ignore `kolibri.*` couldn't get webpack resolver to work nicely with config files.

`eslint-config-prettier` was added to turn off the style-related rules that prettier is supposed to autofix.

### CODE REVIEW

**Comment by rtibbles** [comment]
Unfortunately, this now conflicts with the minor change you had made in your other PR.

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    node:
+      extensions:
+        - .js
+        - .vue
+rules:
+  vue/v-bind-style: 2
+  vue/v-on-style: 2
+  vue/html-quotes: [2, 'double']
+  vue/order-in-components: 2
+  # ignores the 'kolibri.*' imports
```
Comment (rtibbles) [inline .eslintrc:25]: There are webpack extensions that we can use to feed it the aliases and verify those too - but that can wait for another day.

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,26 @@
+parserOptions:
+  ecmaVersion: 7
+  sourceType: module
+extends:
+  - eslint:recommended
+  - prettier
```
Comment (rtibbles) [inline .eslintrc:6]: I assume this makes eslint ignore line length?

Comment (jonboiser) [reply .eslintrc:6]: yeah `eslint-config-prettier` basically turns off every rule that might potentially conflict with `prettier`, including `max-len`.

I tried using `eslint-plugin-prettier`, which flags prettier violations in-editor, but currently there isn't support for Vue components. The violations are supposed to be autofixed anyway.

File: .eslintrc (hunk 1 above)
Comment (jonboiser) [reply .eslintrc:25]: You can configure the import plugin to use webpack module resolution by pointing to the webpack config file, but I had no luck with that approach.

**Pull request merged**